accumulator/pollard.go

```diff
@@ -333,37 +339,39 @@ func (p *Pollard) swapNodes(s arrow, row uint8) (*hashableNode, error) {
 // NOTE errors are not exhaustive; could return garbage without an error
 func (p *Pollard) grabPos(
 	pos uint64) (n, nsib *polNode, hn *hashableNode, err error) {
+	// Grab the tree that the position is at
 	tree, branchLen, bits := detectOffset(pos, p.numLeaves)
-	// fmt.Printf("grab %d, tree %d, bl %d bits %x\n", pos, tree, branchLen, bits)
 	if tree >= uint8(len(p.roots)) {
-		err = fmt.Errorf("want tree %d but only %d trees", tree, len(p.roots))
+		err = ErrorStrings[ErrorNotEnoughTrees]
 		return
 	}
 	n, nsib = &p.roots[tree], &p.roots[tree]
+
 	hn = &hashableNode{dest: n, sib: nsib}
 	for h := branchLen - 1; h != 255; h-- { // go through branch
 		lr := uint8(bits>>h) & 1
+		// grab the sibling of lr
+		lrSib := lr ^ 1
 		if h == 0 { // if at bottom, done
 			hn.dest = nsib // this is kind of confusing eh?
 			hn.sib = n     // but yeah, switch siblingness
-			n, nsib = n.niece[lr^1], n.niece[lr]
+			n, nsib = n.niece[lrSib], n.niece[lr]
 			if nsib == nil || n == nil {
-				// fmt.Printf("gave up ")
 				return // give up and don't make hashable node
```

Comment:
while you're cleaning this up, maybe this comment could be improved. what does "don't make hashable node" mean @adiabat ? it was already made

Reply:
Made improvement to the comment.

Reply:
We could just get rid of the whole `if nsib == nil || n == nil {` lines since they don't do anything.  Once you get to line 358 it's done and returning no matter what.
So the comment is wrong...

Reply:
Removed the branch